@lianghao208 @AxeZhan Hello, I have added a test case named `Override_MultiPoint_plugins_weights_and_avoid_a_plugin_being_loaded_multiple_times`.

Before this fix, it will fail:
```
=== RUN   TestNewFrameworkMultiPointExpansion/Override_MultiPoint_plugins_weights_and_avoid_a_plugin_being_loaded_multiple_times
    framework.go:509: I1129 17:24:29.029130] MultiPoint plugin is explicitly re-configured; overriding plugin="test-plugin"
    framework.go:509: I1129 17:24:29.029173] MultiPoint plugin is explicitly re-configured; overriding plugin="score-plugin-1"
    framework_test.go:899: Unexpected eventToPlugin map (-want,+got):  &config.Plugins{
          	... // 4 identical fields
          	PostFilter: {Enabled: {{Name: "test-plugin"}}},
          	PreScore:   {Enabled: {{Name: "test-plugin"}, {Name: "score-plugin-1"}}},
          	Score: config.PluginSet{
          		Enabled: []config.Plugin{
          			{Name: "score-plugin-1", Weight: 5},
          			{Name: "test-plugin", Weight: 3},
        + 			{Name: "test-plugin", Weight: 3},
          			{Name: "score-plugin-2", Weight: 5},
          		},
          		Disabled: nil,
          	},
          	Reserve: {Enabled: {{Name: "test-plugin"}}},
          	Permit:  {Enabled: {{Name: "test-plugin"}}},
          	... // 4 identical fields
          }
--- FAIL: TestNewFrameworkMultiPointExpansion (0.00s)
    --- PASS: TestNewFrameworkMultiPointExpansion/plugin_expansion (0.00s)
    --- PASS: TestNewFrameworkMultiPointExpansion/disable_MultiPoint_plugin_at_some_extension_points (0.00s)
    --- PASS: TestNewFrameworkMultiPointExpansion/Multiple_MultiPoint_plugins (0.00s)
    --- PASS: TestNewFrameworkMultiPointExpansion/disable_MultiPoint_extension (0.00s)
    --- PASS: TestNewFrameworkMultiPointExpansion/Reorder_MultiPoint_plugins_(specified_extension_takes_precedence) (0.00s)
    --- PASS: TestNewFrameworkMultiPointExpansion/Reorder_MultiPoint_plugins_(specified_extension_only_takes_precedence_when_it_exists_in_MultiPoint) (0.00s)
    --- PASS: TestNewFrameworkMultiPointExpansion/Override_MultiPoint_plugins_weights (0.00s)
    --- PASS: TestNewFrameworkMultiPointExpansion/disable_and_enable_MultiPoint_plugins_with_'*' (0.00s)
    --- PASS: TestNewFrameworkMultiPointExpansion/disable_and_enable_MultiPoint_plugin_by_name (0.00s)
    --- PASS: TestNewFrameworkMultiPointExpansion/Expect_'already_registered'_error (0.00s)
    --- FAIL: TestNewFrameworkMultiPointExpansion/Override_MultiPoint_plugins_weights_and_avoid_a_plugin_being_loaded_multiple_times (0.00s)
```
Also, I have printed the got and want results:
```
tc.wantPlugins.Score: config.PluginSet{Enabled:[]config.Plugin{config.Plugin{Name:"score-plugin-1", Weight:5}, config.Plugin{Name:"test-plugin", Weight:3}, config.Plugin{Name:"score-plugin-2", Weight:5}}, Disabled:[]config.Plugin(nil)}
fw.ListPlugins().Score: config.PluginSet{Enabled:[]config.Plugin{config.Plugin{Name:"score-plugin-1", Weight:5}, config.Plugin{Name:"test-plugin", Weight:3}, config.Plugin{Name:"test-plugin", Weight:3}, config.Plugin{Name:"score-plugin-2", Weight:5}}, Disabled:[]config.Plugin(nil)}
```
We can see from the result, `test-plugin` was loaded 2 times.

After this fix, it will pass:
```
--- PASS: TestNewFrameworkMultiPointExpansion (0.01s)
    --- PASS: TestNewFrameworkMultiPointExpansion/plugin_expansion (0.00s)
    --- PASS: TestNewFrameworkMultiPointExpansion/disable_MultiPoint_plugin_at_some_extension_points (0.00s)
    --- PASS: TestNewFrameworkMultiPointExpansion/Multiple_MultiPoint_plugins (0.00s)
    --- PASS: TestNewFrameworkMultiPointExpansion/disable_MultiPoint_extension (0.00s)
    --- PASS: TestNewFrameworkMultiPointExpansion/Reorder_MultiPoint_plugins_(specified_extension_takes_precedence) (0.00s)
    --- PASS: TestNewFrameworkMultiPointExpansion/Reorder_MultiPoint_plugins_(specified_extension_only_takes_precedence_when_it_exists_in_MultiPoint) (0.00s)
    --- PASS: TestNewFrameworkMultiPointExpansion/Override_MultiPoint_plugins_weights (0.00s)
    --- PASS: TestNewFrameworkMultiPointExpansion/disable_and_enable_MultiPoint_plugins_with_'*' (0.00s)
    --- PASS: TestNewFrameworkMultiPointExpansion/disable_and_enable_MultiPoint_plugin_by_name (0.00s)
    --- PASS: TestNewFrameworkMultiPointExpansion/Expect_'already_registered'_error (0.00s)
    --- PASS: TestNewFrameworkMultiPointExpansion/Override_MultiPoint_plugins_weights_and_avoid_a_plugin_being_loaded_multiple_times (0.00s)
```

--- a/pkg/scheduler/framework/runtime/framework.go
+++ b/pkg/scheduler/framework/runtime/framework.go
@@ -37,6 +37,7 @@ import (
 	"k8s.io/kubernetes/pkg/scheduler/framework"
 	"k8s.io/kubernetes/pkg/scheduler/framework/parallelize"
 	"k8s.io/kubernetes/pkg/scheduler/metrics"
+	"k8s.io/kubernetes/pkg/util/slice"
 )
 
 const (
@@ -526,7 +527,7 @@ func (f *frameworkImpl) expandMultiPointPlugins(logger klog.Logger, profile *con
 		// - part 3: other plugins (excluded by part 1 & 2) in regular extension point.
 		newPlugins := reflect.New(reflect.TypeOf(e.slicePtr).Elem()).Elem()
 		// part 1
-		for _, name := range enabledSet.list {
+		for _, name := range slice.CopyStrings(enabledSet.list) {
 			if overridePlugins.has(name) {
 				newPlugins = reflect.Append(newPlugins, reflect.ValueOf(pluginsMap[name]))
 				enabledSet.delete(name)

--- a/pkg/scheduler/framework/runtime/framework_test.go
+++ b/pkg/scheduler/framework/runtime/framework_test.go
@@ -46,6 +46,7 @@ const (
 	scoreWithNormalizePlugin1         = "score-with-normalize-plugin-1"
 	scoreWithNormalizePlugin2         = "score-with-normalize-plugin-2"
 	scorePlugin1                      = "score-plugin-1"
+	scorePlugin2                      = "score-plugin-2"
 	pluginNotImplementingScore        = "plugin-not-implementing-score"
 	preFilterPluginName               = "prefilter-plugin"
 	preFilterWithExtensionsPluginName = "prefilter-with-extensions-plugin"
@@ -98,6 +99,14 @@ func newScorePlugin1(_ context.Context, injArgs runtime.Object, f framework.Hand
 		return nil, err
 	}
 	return &TestScorePlugin{scorePlugin1, inj}, nil
+}
+
+func newScorePlugin2(_ context.Context, injArgs runtime.Object, f framework.Handle) (framework.Plugin, error) {
+	var inj injectedResult
+	if err := DecodeInto(injArgs, &inj); err != nil {
+		return nil, err
+	}
+	return &TestScorePlugin{scorePlugin2, inj}, nil
 }
 
 func newPluginNotImplementingScore(_ context.Context, _ runtime.Object, _ framework.Handle) (framework.Plugin, error) {
@@ -358,11 +367,13 @@ func (t TestBindPlugin) Bind(ctx context.Context, state *framework.CycleState, p
 	return nil
 }
 
+// nolint:errcheck   // Ignore the error returned by Register as before
 var registry = func() Registry {
 	r := make(Registry)
 	r.Register(scoreWithNormalizePlugin1, newScoreWithNormalizePlugin1)
 	r.Register(scoreWithNormalizePlugin2, newScoreWithNormalizePlugin2)
 	r.Register(scorePlugin1, newScorePlugin1)
+	r.Register(scorePlugin2, newScorePlugin2)
 	r.Register(pluginNotImplementingScore, newPluginNotImplementingScore)
 	r.Register(duplicatePluginName, newDuplicatePlugin)
 	r.Register(testPlugin, newTestPlugin)
@@ -825,6 +836,44 @@ func TestNewFrameworkMultiPointExpansion(t *testing.T) {
 				},
 			},
 			wantErr: "already registered",
+		},
+		{
+			name: "Override MultiPoint plugins weights and avoid a plugin being loaded multiple times",
+			plugins: &config.Plugins{
+				MultiPoint: config.PluginSet{
+					Enabled: []config.Plugin{
+						{Name: testPlugin},
+						{Name: scorePlugin1},
+					},
+				},
+				Score: config.PluginSet{
+					Enabled: []config.Plugin{
+						{Name: scorePlugin1, Weight: 5},
+						{Name: scorePlugin2, Weight: 5},
+						{Name: testPlugin, Weight: 3},
+					},
+				},
+			},
+			wantPlugins: &config.Plugins{
+				QueueSort:  config.PluginSet{Enabled: []config.Plugin{{Name: testPlugin}}},
+				PreFilter:  config.PluginSet{Enabled: []config.Plugin{{Name: testPlugin}}},
+				Filter:     config.PluginSet{Enabled: []config.Plugin{{Name: testPlugin}}},
+				PostFilter: config.PluginSet{Enabled: []config.Plugin{{Name: testPlugin}}},
+				PreScore: config.PluginSet{Enabled: []config.Plugin{
+					{Name: testPlugin},
+					{Name: scorePlugin1},
+				}},
+				Score: config.PluginSet{Enabled: []config.Plugin{
+					{Name: scorePlugin1, Weight: 5},
+					{Name: testPlugin, Weight: 3},
+					{Name: scorePlugin2, Weight: 5},
+				}},
+				Reserve:  config.PluginSet{Enabled: []config.Plugin{{Name: testPlugin}}},
+				Permit:   config.PluginSet{Enabled: []config.Plugin{{Name: testPlugin}}},
+				PreBind:  config.PluginSet{Enabled: []config.Plugin{{Name: testPlugin}}},
+				Bind:     config.PluginSet{Enabled: []config.Plugin{{Name: testPlugin}}},
+				PostBind: config.PluginSet{Enabled: []config.Plugin{{Name: testPlugin}}},
+			},
 		},
 	}
 

